### DISCUSSION TITLE
How to update relation fields automatically?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#1111] [Q&A] [answered]
I have a `post` collection. `post` collection has a relation field named `comments`. At the moment of creating a record in the `post` collection, it won't have any comment. After a moment of creating a record in the `post` collection, it might have a comment. When a record in the `comments` collection is created, my `post` collection should be updated automatically.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I fully understand the question.

PocketBase supports 2 type of relations - single and multiple. From your example it seems that you have multi-relation `comments` field in the `posts` collection, right? If that's the case, when creating a new comment, you have to manually attach it to the related `posts` record by sending an update `posts` request.

To avoid this, the better approach in this situation would be to reverse the relation, aka. delete the `comments` relation field from your `posts` collection and instead add a nonempty single `post` relation field in the `comments` collection. This way, when you create a new comment it would be automatically attached to the `posts` record.

Later, to get the related `comments` to a `posts` record using `?expand` you can make use of the indirect expansion syntax, aka.:

```js
pb.collection("posts").getOne("POST_ID", { expand: 'comments(post)' });
```
_(you can find more information on expand in https://pocketbase.io/docs/expanding-relations/; the indirect expand example illustrates similar posts-comments relationship - https://pocketbase.io/docs/expanding-relations/#indirect-expand )_

**Reply by Mr-M1M3** [reply]
If I reverse the relation, how would I create a record in the post collection? A post might not have a comment but a comment have to have a post.

**Reply by ganigeorgiev** [reply]
> If I reverse the relation, how would I create a record in the post collection?

With just the `posts` collection fields, eg.:

```js
await pb.collection("posts").create({
    title: "Some title",
    description: "Some description...",
})
```

And then later if you want to add a comment to a post:

```js
await pb.collection("comments").create({
    post: "POST_ID",
    message: "Some message...",
})
```

**Reply by Mr-M1M3** [reply]
> With just the posts collection fields...

A comment should not be empty right? I would probably add a `nonempty` field.

**Reply by ganigeorgiev** [reply]
Yes, the `post` relation field in your `comments` collection should be marked as nonempty.

**Reply by Mr-M1M3** [reply]
> Yes, the `post` relation field in your `comments` collection should be marked as nonempty.

what about comment body?

**Reply by ganigeorgiev** [reply]
Yes, that make sense. It's up to you what fields and constraints a comment should have.

**Reply by Mr-M1M3** [reply]
Now, if I reverse the relation, I would have to provide a comment body. But, when I create a post, I won't have any comments.  Also, a single post might not have a comment.

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand what do you mean. Isn't that the desired behavior?

**Reply by Mr-M1M3** [reply]
Actually, no. I have edited the question. Do you understand now?

**Reply by ganigeorgiev** [reply]
@Mr-M1M3 Your edited question is inline with what I initially understood. As mentioned above, if the `comments` relation is in the `posts` collection then you'll have to manually update it. That's why I suggested to reverse the relation, aka. delete the `comments` field from your `posts` collection and add a `post` field in your `comments` collection. Please reread the answer again.

The expand example structure also is very similar to what you want - https://pocketbase.io/docs/expanding-relations/

**Reply by Mr-M1M3** [reply]
I think, I will just have to update it manually. Reverse relation could create some problems. Thanks BTW. You are the most helpful person on the internet out there. Thanks.

**Reply by ganigeorgiev** [reply]
> Reverse relation could create some problems.

It's the opposite relation that could create problem because you'll allow any user to change the posts comments when submitting a new one.

Please reread carefully again the answer and the comments after it.

If you DON'T reverse the relation and keep it as it is, then you'll have to submit all comment ids to the `posts` update call:

```js
// create a new comment
const comment = await pb.collection("comments").create({
    message: "Some message...",
})

await pb.collection("posts").update("POST_ID", {
    title: "Some title",
    description: "Some description...",
    comments: ["COMMENT_ID1", "COMMENT_ID2", comment.id]
})
```

But if you reverse the relation, you can have:

```js
// create post
await pb.collection("posts").create({
    title: "Some title",
    description: "Some description...",
})

// and then later when you want to add a new comment
await pb.collection("comments").create({
    post: "POST_ID",
    message: "Some message...",
})
```

**Reply by Mr-M1M3** [reply]
I think, I have got what you meant by reversing the relation. I was just dumb. I thought, if I reverse the relation, I would have to create a post by creating a comment  (by passing `post` collection's data to comment's post field). Now I understand, I can create a post separately even after reversing a collection and I can also get expected behavior by using `indirect expansion` . How dumb I am !!!  Thanks a lot mate...
